# Cleanig the 2015 BRFSS Survey Data from the CDC




Data source: https://www.cdc.gov/brfss/annual_data/annual_2015.html

This is the data from the Behavioral Risk Factor Surveillance System from 2015 published annually by the Center of Disease Control (CDC).

This is our take on cleaning the data so it can be easialy be used for any kind of ML.

Obviously, this isn't the way to go though if you what to find new and interesting patterns and correlations between features because we hand-picked the features ourselves. 

###  Required Libraries 

In [1]:
import os
import pandas as pd

In [2]:
#Import data from CDC BRFSS(2015)
pd.set_option("display.max_columns", 330)
df = pd.read_csv('Datasets/BRFSS_2015.csv')

In [3]:
# Original shape
df.shape

(441456, 330)

The data coontains answers to 330 questions of 441.456 people all across the United States

In [4]:
df.head()

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE         SEQNO  \
0     1.0     1.0  b'01292015'  b'01'  b'29'  b'2015'    1200.0  2.015000e+09   
1     1.0     1.0  b'01202015'  b'01'  b'20'  b'2015'    1100.0  2.015000e+09   
2     1.0     1.0  b'02012015'  b'02'  b'01'  b'2015'    1200.0  2.015000e+09   
3     1.0     1.0  b'01142015'  b'01'  b'14'  b'2015'    1100.0  2.015000e+09   
4     1.0     1.0  b'01142015'  b'01'  b'14'  b'2015'    1100.0  2.015000e+09   

           _PSU  CTELENUM  PVTRESD1  COLGHOUS  STATERES  CELLFON3  LADULT  \
0  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
1  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
2  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
3  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
4  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   

   NUMADULT        NUMMEN  NUMWOMEN  CTELNUM1  CELLFON2  CADULT  PVTRESD2  \
0       3.0  1.000000e+00       2.0       NaN       NaN     NaN       NaN   
1       1.0  5.397605e-79       1.0       NaN       NaN     NaN       NaN   
2       2.0  1.000000e+00       1.0       NaN       NaN     NaN       NaN   
3       3.0  1.000000e+00       2.0       NaN       NaN     NaN       NaN   
4       2.0  1.000000e+00       1.0       NaN       NaN     NaN       NaN   

   CCLGHOUS  CSTATE  LANDLINE  HHADULT  GENHLTH  PHYSHLTH  MENTHLTH  POORHLTH  \
0       NaN     NaN       NaN      NaN      5.0      15.0      18.0      10.0   
1       NaN     NaN       NaN      NaN      3.0      88.0      88.0       NaN   
2       NaN     NaN       NaN      NaN      4.0      15.0      88.0      88.0   
3       NaN     NaN       NaN      NaN      5.0      30.0      30.0      30.0   
4       NaN     NaN       NaN      NaN      5.0      20.0      88.0      30.0   

   HLTHPLN1  PERSDOC2  MEDCOST  CHECKUP1  BPHIGH4  BPMEDS  BLOODCHO  CHOLCHK  \
0       1.0       1.0      2.0       1.0      1.0     1.0       1.0      1.0   
1       2.0       1.0      1.0       4.0      3.0     NaN       1.0      4.0   
2       1.0       2.0      2.0       1.0      3.0     NaN       1.0      1.0   
3       1.0       2.0      1.0       1.0      1.0     1.0       1.0      1.0   
4       1.0       1.0      2.0       1.0      3.0     NaN       1.0      1.0   

   TOLDHI2  CVDINFR4  CVDCRHD4  CVDSTRK3  ASTHMA3  ASTHNOW  CHCSCNCR  \
0      1.0       2.0       2.0       2.0      1.0      1.0       2.0   
1      2.0       2.0       2.0       2.0      2.0      NaN       2.0   
2      1.0       7.0       2.0       1.0      2.0      NaN       2.0   
3      1.0       2.0       2.0       2.0      2.0      NaN       2.0   
4      2.0       2.0       2.0       2.0      2.0      NaN       2.0   

   CHCOCNCR  CHCCOPD1  HAVARTH3  ADDEPEV2  CHCKIDNY  DIABETE3  DIABAGE2  SEX  \
0       2.0       1.0       1.0       1.0       2.0       3.0       NaN  2.0   
1       2.0       2.0       2.0       2.0       2.0       3.0       NaN  2.0   
2       1.0       2.0       1.0       2.0       2.0       3.0       NaN  2.0   
3       1.0       2.0       1.0       1.0       2.0       3.0       NaN  2.0   
4       2.0       2.0       1.0       2.0       2.0       3.0       NaN  2.0   

   MARITAL  EDUCA  RENTHOM1  NUMHHOL2  NUMPHON2  CPDEMO1  VETERAN3  EMPLOY1  \
0      1.0    4.0       1.0       2.0       NaN      1.0       2.0      8.0   
1      2.0    6.0       1.0       2.0       NaN      2.0       2.0      3.0   
2      2.0    4.0       1.0       2.0       NaN      1.0       2.0      7.0   
3      1.0    4.0       1.0       2.0       NaN      1.0       2.0      8.0   
4      1.0    5.0       1.0       2.0       NaN      2.0       2.0      8.0   

   CHILDREN  INCOME2  INTERNET  WEIGHT2  HEIGHT3  PREGNANT  QLACTLM2  \
0      88.0      3.0       2.0    280.0    510.0       NaN       1.0   
1      88.0      1.0       1.0    165.0    508.0       NaN       1.0   
2      88.0     9

In [5]:
df.tail()

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE  \
441451    72.0    11.0  b'12162015'  b'12'  b'16'  b'2015'    1100.0   
441452    72.0    11.0  b'12142015'  b'12'  b'14'  b'2015'    1100.0   
441453    72.0    11.0  b'12232015'  b'12'  b'23'  b'2015'    1200.0   
441454    72.0    11.0  b'12152015'  b'12'  b'15'  b'2015'    1100.0   
441455    72.0    11.0  b'12152015'  b'12'  b'15'  b'2015'    1100.0   

               SEQNO          _PSU  CTELENUM  PVTRESD1  COLGHOUS  STATERES  \
441451  2.015005e+09  2.015005e+09       NaN       NaN       NaN       NaN   
441452  2.015005e+09  2.015005e+09       NaN       NaN       NaN       NaN   
441453  2.015005e+09  2.015005e+09       NaN       NaN       NaN       NaN   
441454  2.015005e+09  2.015005e+09       NaN       NaN       NaN       NaN   
441455  2.015005e+09  2.015005e+09       NaN       NaN       NaN       NaN   

        CELLFON3  LADULT  NUMADULT  NUMMEN  NUMWOMEN  CTELNUM1  CELLFON2  \
441451       NaN     NaN       NaN     NaN       NaN       1.0       1.0   
441452       NaN     NaN       NaN     NaN       NaN       1.0       1.0   
441453       NaN     NaN       NaN     NaN       NaN       1.0       1.0   
441454       NaN     NaN       NaN     NaN       NaN       1.0       1.0   
441455       NaN     NaN       NaN     NaN       NaN       1.0       1.0   

        CADULT  PVTRESD2  CCLGHOUS  CSTATE  LANDLINE  HHADULT  GENHLTH  \
441451     2.0       1.0       NaN     1.0       2.0      4.0      4.0   
441452     2.0       1.0       NaN     1.0       2.0      2.0      1.0   
441453     2.0       1.0       NaN     1.0       2.0      2.0      4.0   
441454     1.0       1.0       NaN     1.0       1.0      3.0      3.0   
441455     2.0       1.0       NaN     1.0       2.0      2.0      2.0   

        PHYSHLTH  MENTHLTH  POORHLTH  HLTHPLN1  PERSDOC2  MEDCOST  CHECKUP1  \
441451      88.0      88.0       NaN       1.0       1.0      2.0       1.0   
441452      88.0      88.0       NaN       1.0       1.0      2.0       1.0   
441453      88.0      20.0      88.0       1.0       2.0      2.0       1.0   
441454      88.0      88.0       NaN       1.0       1.0      2.0       1.0   
441455      88.0      88.0       NaN       1.0       1.0      2.0       1.0   

        BPHIGH4  BPMEDS  BLOODCHO  CHOLCHK  TOLDHI2  CVDINFR4  CVDCRHD4  \
441451      1.0     1.0       1.0      1.0      1.0       2.0       2.0   
441452      3.0     NaN       1.0      1.0      2.0       2.0       2.0   
441453      1.0     1.0       1.0      1.0      1.0       2.0       2.0   
441454      1.0     1.0       1.0      1.0      2.0       2.0       2.0   
441455      1.0     1.0       1.0      1.0      1.0       1.0       2.0   

        CVDSTRK3  ASTHMA3  ASTHNOW  CHCSCNCR  CHCOCNCR  CHCCOPD1  HAVARTH3  \
441451       2.0      2.0      NaN       2.0       2.0       2.0       1.0   
441452       2.0      2.0      NaN       2.0       2.0       2.0       2.0   
441453       2.0      2.0      NaN       2.0       1.0       2.0       2.0   
441454       2.0      2.0      NaN       2.0       2.0       2.0       2.0   
441455       2.0      2.0      NaN       1.0       2.0       1.0       2.0   

        ADDEPEV2  CHCKIDNY  DIABETE3  DIABAGE2  SEX  MARITAL  EDUCA  RENTHOM1  \
441451       2.0       2.0       1.0      55.0  2.0      3.0    2.0       1.0   
441452       2.0       2.0       3.0       NaN  2.0      1.0    5.0       1.0   
441453       2.0       2.0       3.0       NaN  2.0      1.0    4.0       1.0   
441454       2.0       2.0       3.0       NaN  1.0      5.0    5.0       1.0   
441455       2.0       2.0       1.0      45.0  2.0      1.0    6.0       1.0   

        NUMHHOL2  NUMPHON2  CPDEMO1  VETERAN3  EMPLOY1  CHILDREN  INCOME2  \
441451       NaN       NaN      NaN       2.0      7.0      88.0      4.0   
441452       NaN       NaN      NaN       2.0      1.0       1.0      2.0   
441453       NaN       NaN      NaN       2.0      7.0      88.0      5.0   
441454       N

In [6]:
#checking for NaN values
num_NaN = df.isna().sum()
total_NaN = df.isna().sum().sum()
print(total_NaN)

64809261


In total there are 64.809.261 NaN values.

In [7]:
total_values = df.shape[0] * df.shape[1]
percent_NaN = total_NaN / total_values * 100
print('{:.2f}% of the values in the data are NaN'.format(percent_NaN))

44.49% of the values in the data are NaN


### Selecting specific columns related to heart diseases

In [9]:
df_selected = df[['_MICHD','_AGEG5YR', 'SEX', 'EDUCA', 'CHILDREN', '_TOTINDA', 'VETERAN3', 'EMPLOY1', '_STATE',
                  'INCOME2', 'CHECKUP1', 'RENTHOM1', 'MSCODE', '_PRACE1', 'MARITAL', 'SMOKE100', '_BMI5', '_RFBMI5',
                  '_RFHYPE5', 'TOLDHI2', '_CHOLCHK', 'CVDSTRK3', 'DIABETE3', '_FRTLT1', '_VEGLT1', '_RFDRHV5', 
                  'HLTHPLN1', 'MEDCOST', 'GENHLTH', 'MENTHLTH', 'PHYSHLTH',  'DIFFWALK', 'CHCCOPD1', 'MAXVO2_', 
                  '_PACAT1', '_PAINDX1']]


In [10]:
df_selected.head()

_MICHD  _AGEG5YR  SEX  EDUCA  CHILDREN  _TOTINDA  VETERAN3  EMPLOY1  \
0     2.0       9.0  2.0    4.0      88.0       2.0       2.0      8.0   
1     2.0       7.0  2.0    6.0      88.0       1.0       2.0      3.0   
2     NaN      11.0  2.0    4.0      88.0       9.0       2.0      7.0   
3     2.0       9.0  2.0    4.0       1.0       2.0       2.0      8.0   
4     2.0       9.0  2.0    5.0      88.0       2.0       2.0      8.0   

   _STATE  INCOME2  CHECKUP1  RENTHOM1  MSCODE  _PRACE1  MARITAL  SMOKE100  \
0     1.0      3.0       1.0       1.0     3.0      1.0      1.0       1.0   
1     1.0      1.0       4.0       1.0     5.0      1.0      2.0       1.0   
2     1.0     99.0       1.0       1.0     5.0      1.0      2.0       NaN   
3     1.0      8.0       1.0       1.0     3.0      1.0      1.0       2.0   
4     1.0     77.0       1.0       1.0     3.0      1.0      1.0       2.0   

    _BMI5  _RFBMI5  _RFHYPE5  TOLDHI2  _CHOLCHK  CVDSTRK3  DIABETE3  _FRTLT1  \
0  4018.0      2.0       2.0      1.0       1.0       2.0       3.0      2.0   
1  2509.0      2.0       1.0      2.0       2.0       2.0       3.0      2.0   
2  2204.0      1.0       1.0      1.0       1.0       1.0       3.0      9.0   
3  2819.0      2.0       2.0      1.0       1.0       2.0       3.0      1.0   
4  2437.0      1.0       1.0      2.0       1.0       2.0       3.0      9.0   

   _VEGLT1  _RFDRHV5  HLTHPLN1  MEDCOST  GENHLTH  MENTHLTH  PHYSHLTH  \
0      1.0       1.0       1.0      2.0      5.0      18.0      15.0   
1      2.0       1.0       2.0      1.0      3.0      88.0      88.0   
2      9.0       9.0       1.0      2.0      4.0      88.0      15.0   
3      2.0       1.0       1.0      1.0      5.0      30.0      30.0   
4      1.0       1.0       1.0      2.0      5.0      88.0      20.0   

   DIFFWALK  CHCCOPD1  MAXVO2_  _PACAT1  _PAINDX1  
0       1.0       1.0   2469.0      4.0       2.0  
1       2.0       2.0   2876.0      2.0       1.0  
2       NaN       2.0   2173.0      9.0       9.0  
3       1.0       2.0   2469.0      4.0       2.0  
4       2.0       2.0   2543.0      4.0       2.0

In [11]:
# Selected heart attack/disease relaated columns 
df_selected.shape

(441456, 36)

In [12]:
# Checking how many of the remaining features have NaN values
df_selected.isnull().sum(axis = 0)

_MICHD        3942
_AGEG5YR         0
SEX              0
EDUCA            0
CHILDREN         5
_TOTINDA         0
VETERAN3         6
EMPLOY1          0
_STATE           0
INCOME2       3301
CHECKUP1         1
RENTHOM1         0
MSCODE      190877
_PRACE1          0
MARITAL          0
SMOKE100     14255
_BMI5        36398
_RFBMI5          0
_RFHYPE5         0
TOLDHI2      59154
_CHOLCHK         0
CVDSTRK3         0
DIABETE3         7
_FRTLT1          0
_VEGLT1          0
_RFDRHV5         0
HLTHPLN1         0
MEDCOST          1
GENHLTH          2
MENTHLTH         0
PHYSHLTH         1
DIFFWALK     12334
CHCCOPD1         0
MAXVO2_          0
_PACAT1          0
_PAINDX1         0
dtype: int64

In [13]:
num_NaN = df_selected.isna().sum()
total_NaN = df_selected.isna().sum().sum()
print(total_NaN)
total_values = df_selected.shape[0] * df_selected.shape[1]
percent_NaN = total_NaN / total_values * 100
print('{:.2f}% of the values in the data are NaN'.format(percent_NaN))

320284
2.02% of the values in the data are NaN


In [14]:
#droping NaN vaules
df_selected.dropna(inplace=True)

/Users/j.christiansen/opt/anaconda3/lib/python3.9/site-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


In [15]:
df_selected.groupby(['_MICHD']).size()

_MICHD
1.0     24099
2.0    182153
dtype: int64

In [16]:
# Intermidate shape (just after dropping NaN values)
df_selected.shape

(206252, 36)

First remoddeling leaves us with 206252 rows and 36 columns.

### Modifing and cleaning all the values

Now we are cleaning each of the single features: converting them to proper scales (0/1) and droping additional values of no interest (Don’t know/Not Sure/Refused/Missing).

In [17]:
# 1. var (_MICHD)
# Respondents that have ever reported having coronary heart disease (CHD) or myocardial infarction (MI)
# Transforming our traget value of ever having had a heart attack or disease
# Changing 2 to 0 because this means someone did not have MI (myocardial infarction) or CHD (coronary heart disease)

df_selected['_MICHD'] = df_selected['_MICHD'].replace({2: 0})
df_selected._MICHD.unique()

/var/folders/4x/5_y4_03j3b5bsz_s2sq74dcr0000gn/T/ipykernel_3356/2782479277.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected['_MICHD'] = df_selected['_MICHD'].replace({2: 0})


array([0., 1.])

In [18]:
# 2. var (_AGEG5YR) Fourteen-level age category
# Age is already ordinal; 1 equals ages ranging from 18-24 all the way up to 13 which equals to age 80 and older
# The scale increments by 5 years 
# Remove 14 because it is don't know or missing values

df_selected = df_selected[df_selected._AGEG5YR != 14]
df_selected._AGEG5YR.unique()

array([ 9.,  7., 11., 10., 13., 12.,  8.,  4.,  6.,  2.,  5.,  1.,  3.])

In [19]:
# 3. var (SEX) Indicate sex of respondent
# Changing the scale of sex to 0,1 --> 2 to 0; Female as 0, Male is 1

df_selected['SEX'] = df_selected['SEX'].replace({2:0})
df_selected.SEX.unique()

array([0., 1.])

In [20]:
# 4. var (EDUCA)
# What is the highest grade or year of school you completed?
# EDUCA is already an ordinal variable on a scale from 1-6 (1 being never attended school or kindergarten; 
# 6 being college for 4 years or more (graduate)
# Remove 9 for missing/refused values

df_selected = df_selected[df_selected.EDUCA != 9]
df_selected.EDUCA.unique()

array([4., 6., 5., 3., 2., 1.])

In [21]:
# 5. var (CHILDREN)
# How many children less than 18 years of age live in your household? Scale 1-87, 88 = None
# Remove 99 refused values / transforming 88 to 0

df_selected = df_selected[df_selected.CHILDREN != 99]
df_selected['CHILDREN'] = df_selected['CHILDREN'].replace({88:0})
df_selected.CHILDREN.unique()

array([ 0.,  1.,  2.,  3.,  4.,  5.,  6., 12.,  7.,  8., 10., 32.,  9.,
       20., 23., 41., 77.])

In [22]:
# 6. var (_TOTINDA)
# Adults who reported doing physical activity or exercise during the past 30 days other than their regular job
# Changing the scale to 0,1 --> 2 to 0; No exercise as 0, Exercise as 1
# Remove 9 refused values

df_selected = df_selected[df_selected._TOTINDA != 9]
df_selected['_TOTINDA'] = df_selected['_TOTINDA'].replace({2:0})
df_selected._TOTINDA.unique()

array([0., 1.])

In [23]:
# 7. var (VETERAN3) Have you ever served on active duty in the United States Armed Forces, 
                  # either in the regular military or in a National Guard or military reserve unit? 
# Changing the scale of Veteran to 0,1 --> 2 to 0; Non-Veteran as 0, Veteran as 1

df_selected = df_selected[df_selected.VETERAN3 != 7]
df_selected = df_selected[df_selected.VETERAN3 != 9]
df_selected['VETERAN3'] = df_selected['VETERAN3'].replace({2:0})
df_selected.VETERAN3.unique()

array([0., 1.])

In [24]:
# 8. var (EMPLOY1) Employment status
# EMPLOY1 is already an ordinal variable on a scale from 1-8 (1 being employed for wages; 8 being unable to work)
# Remove 9 for refused values

df_selected = df_selected[df_selected.EMPLOY1 != 9]
df_selected.EMPLOY1.unique()

array([8., 3., 2., 7., 5., 1., 4., 6.])

In [25]:
# 9. var (_STATE) State FIPS Code
# no change needed

In [26]:
# 10. var (INCOME2) Is your annual household income from all sources: 
# INCOME2 is already ordinal variable on a scale from 1-8 (1 being less than $10,000; 8 being $75,000 or more)
# Remove 77 and 99 for don't know and refused values 

df_selected = df_selected[df_selected.INCOME2 != 77]
df_selected = df_selected[df_selected.INCOME2 != 99]
df_selected.INCOME2.unique()

array([3., 1., 8., 6., 4., 7., 2., 5.])

In [27]:
# 11. var (CHECKUP1)
# About how long has it been since you last visited a doctor for a routine checkup?
# CHECKUP1 is already ordinal variable on a scale from 1-8 (1 being within a yesr; 8 being never)
# Remove 9 for refused values 

df_selected = df_selected[df_selected.CHECKUP1 != 9]
df_selected.CHECKUP1.unique()

array([1., 4., 2., 3., 8., 7.])

In [28]:
# 12. var (RENTHOM1) --> OWNHOME
# RENTHOM1 is already an ordinal variable 
# Changing the scale of RENTHOM1 to 0,1 --> 2 to 0; Rent home as 0, Own home is 1
# Remove 3, 7, 9 for other arrangements, don't know or refused values

df_selected = df_selected[df_selected.RENTHOM1 != 3]
df_selected = df_selected[df_selected.RENTHOM1 != 7]
df_selected = df_selected[df_selected.RENTHOM1 != 9]
df_selected['RENTHOM1'] = df_selected['RENTHOM1'].replace({2:0})
df_selected.RENTHOM1.unique()

array([1., 0.])

In [29]:
# 13. var (MSCODE) Metropolian Status Code
# no changes needed

In [30]:
# 14. var (_PRACE1) Preferred race category
# _PRACE1 is already an ordinal variable 
# Remove 7, 8, 77, 99 for No preferred race, Multiracial but preferred race not answered, don't know or refused values

df_selected = df_selected[df_selected._PRACE1 != 7]
df_selected = df_selected[df_selected._PRACE1 != 8]
df_selected = df_selected[df_selected._PRACE1 != 77]
df_selected = df_selected[df_selected._PRACE1 != 99]
df_selected._PRACE1.unique()

array([1., 2., 3., 6., 4., 5.])

In [31]:
# 15. var (MARITAL) marital status
# MARITAL is already an ordinal variable 
# Remove 9 for refused values

df_selected = df_selected[df_selected.MARITAL != 9]
df_selected.MARITAL.unique()

array([1., 2., 3., 6., 5., 4.])

In [32]:
# 16. var (SMOKE100) Smoked 100 cigarettes in life?
# SMOKE100 is already an ordinal variable 
# Changing the scale of SMOKE100 to 0,1 --> 2 to 0; No as 0, Yes is 1
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.SMOKE100 != 7]
df_selected = df_selected[df_selected.SMOKE100 != 9]
df_selected['SMOKE100'] = df_selected['SMOKE100'].replace({2:0})
df_selected.SMOKE100.unique()

array([1., 0.])

In [33]:
# 17. var (_BMI5) Body Mass Index (BMI)
# These BMI are multiplied by 100. So e.g. a BMI of 4018 is really 40.18

df_selected['_BMI5'] = df_selected['_BMI5'].div(100).round(0)
df_selected._BMI5.unique()

array([40., 25., 28., 27., 24., 30., 34., 31., 23., 26., 33., 21., 22.,
       38., 19., 32., 37., 36., 29., 20., 35., 45., 39., 47., 18., 43.,
       55., 49., 42., 17., 16., 41., 44., 50., 59., 52., 46., 54., 53.,
       48., 57., 51., 15., 56., 14., 13., 58., 66., 62., 85., 72., 61.,
       63., 67., 65., 64., 60., 70., 81., 12., 69., 82., 76., 73., 75.,
       77., 68., 87., 89., 71., 80., 90., 84., 74., 79., 97., 88.])

In [34]:
# 18. var (_RFBMI5) obease/overweight ?
# Adults who have a body mass index greater than 25.00 (Overweight or Obese)
# _RFBMI5 is already an ordinal variable 
# Changing the scale of _RFBMI5 to 0,1 --> 2 to 0; No as 0, Yes is 1

df_selected['_RFBMI5'] = df_selected['_RFBMI5'].replace({1:0})
df_selected['_RFBMI5'] = df_selected['_RFBMI5'].replace({2:1})
df_selected._RFBMI5.unique()

array([1., 0.])

In [35]:
# 19. var (_RFHYPE5)
# Adults who have been told they have high blood pressure by a doctor
# Changing the scale of _RFHYPE5 to 0,1; No as 0, Yes is 1

df_selected['_RFHYPE5'] = df_selected['_RFHYPE5'].replace({1:0})
df_selected['_RFHYPE5'] = df_selected['_RFHYPE5'].replace({2:1})
df_selected._RFBMI5.unique()

array([1., 0.])

In [36]:
# 20. var (TOLDHI2)
# Have you EVER been told by a doctor, nurse or other health professional that your blood cholesterol is high?
# Changing the scale of TOLDHI2 to 0,1; No as 0, Yes is 1
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.TOLDHI2 != 7]
df_selected = df_selected[df_selected.TOLDHI2 != 9]
df_selected['TOLDHI2'] = df_selected['TOLDHI2'].replace({2:0})
df_selected.TOLDHI2.unique()

array([1., 0.])

In [37]:
# 21. var (_CHOLCHK)
# Cholesterol check within past five years
# _CHOLCHK is already ordinal variable on a scale from 1-2 (1 being yes; 2 being no)
# Changing the scale of _CHOLCHK to 0,1; No as 0, Yes is 1
# Remove 9 for don't know or refused values

df_selected = df_selected[df_selected._CHOLCHK != 9]
df_selected['_CHOLCHK'] = df_selected['_CHOLCHK'].replace({2:0})
df_selected._CHOLCHK.unique()

array([1., 0.])

In [38]:
# 22. var (CVDSTRK3)
# (Ever told) you had a stroke
# Changing the scale of CVDSTRK3 to 0,1; No as 0, Yes is 1
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.CVDSTRK3 != 7]
df_selected = df_selected[df_selected.CVDSTRK3 != 9]
df_selected['CVDSTRK3'] = df_selected['CVDSTRK3'].replace({2:0})
df_selected.CVDSTRK3.unique()

array([0., 1.])

In [39]:
# 23. var (DIABETE3)
# (Ever told) you have diabetes
# Changing the scale of DIABETE3 to 0,1; No as 0, Yes is 1 (counting pre-diabetes/borderline diabetes as diabetes)
# Remove 2(during pregnancy), 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.DIABETE3 != 2]
df_selected = df_selected[df_selected.DIABETE3 != 7]
df_selected = df_selected[df_selected.DIABETE3 != 9]
df_selected['DIABETE3'] = df_selected['DIABETE3'].replace({3:0})
df_selected['DIABETE3'] = df_selected['DIABETE3'].replace({4:1})
df_selected.DIABETE3.unique()

array([0., 1.])

In [40]:
# 24. var (_FRTLT1)
# Consume Fruit 1 or more times per day
# Changing the scale of _FRTLT1 to 0,1; No as 0, Yes is 1 
# Remove 9 for don't know or refused values

df_selected = df_selected[df_selected._FRTLT1 != 9]
df_selected['_FRTLT1'] = df_selected['_FRTLT1'].replace({2:0})
df_selected.DIABETE3.unique()

array([0., 1.])

In [41]:
# 25. var (_VEGLT1)
# Consume Vegetables 1 or more times per day
# Changing the scale of _VEGLT1 to 0,1; No as 0, Yes is 1 
# Remove 9 for don't know or refused values

df_selected = df_selected[df_selected._VEGLT1 != 9]
df_selected['_VEGLT1'] = df_selected['_VEGLT1'].replace({2:0})
df_selected._VEGLT1.unique()

array([1., 0.])

In [42]:
# 26. var (_RFDRHV5)
# Heavy drinkers (adult men having more than 14 drinks per week)
# Changing the scale of _RFDRHV5 to 0,1; No as 0, Yes is 1 
# Remove 9 for don't know or refused values

df_selected['_RFDRHV5'] = df_selected['_RFDRHV5'].replace({1:0})
df_selected['_RFDRHV5'] = df_selected['_RFDRHV5'].replace({2:1})
df_selected = df_selected[df_selected._RFDRHV5 != 9]
df_selected._RFBMI5.unique()

array([1., 0.])

In [43]:
# 27. var (HLTHPLN1)
# Do you have any kind of health care coverage ?
# Changing the scale of HLTHPLN1 to 0,1; No as 0, Yes is 1 
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.HLTHPLN1 != 7]
df_selected = df_selected[df_selected.HLTHPLN1 != 9]
df_selected['HLTHPLN1'] = df_selected['HLTHPLN1'].replace({2:0})
df_selected.HLTHPLN1.unique()

array([1., 0.])

In [44]:
# 28. var (MEDCOST) 
# Was there a time in the past 12 months when you needed to see a doctor but could not because of cost?
# Changing the scale of MEDCOST to 0,1; No as 0, Yes is 1 
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.MEDCOST != 7]
df_selected = df_selected[df_selected.MEDCOST != 9]
df_selected['MEDCOST'] = df_selected['MEDCOST'].replace({2:0})
df_selected.MEDCOST.unique()

array([0., 1.])

In [45]:
# 29. var (GENHLTH) 
# Would you say that in general your health is: 1-5 scale, 1 being the best
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.GENHLTH != 7]
df_selected = df_selected[df_selected.GENHLTH != 9]
df_selected.GENHLTH.unique()

array([5., 3., 2., 4., 1.])

In [46]:
# 30. var (MENTHLTH) 
# Mental health, including stress & depression, out of 30 days for how many days was your mental health not good?
# change 88 to 0 for NONE
# Remove 77, 99 for don't know or refused values

df_selected = df_selected[df_selected.MENTHLTH != 77]
df_selected = df_selected[df_selected.MENTHLTH != 99]
df_selected['MENTHLTH'] = df_selected['MENTHLTH'].replace({88:0})
df_selected.MENTHLTH.unique()

array([18.,  0., 30.,  3.,  5., 15., 10.,  6., 20.,  2., 25.,  1., 29.,
        4.,  8., 21., 14., 26.,  7., 16., 28., 11., 12., 24., 17., 13.,
       19., 22., 27.,  9., 23.])

In [47]:
# 31. var (PHYSHLTH)
# physical health, including physical illness & injury, out of 30 days for how many was your physical health not good?
# change 88 to 0 for NONE
# Remove 77, 99 for don't know or refused values

df_selected = df_selected[df_selected.PHYSHLTH != 77]
df_selected = df_selected[df_selected.PHYSHLTH != 99]
df_selected['PHYSHLTH'] = df_selected['PHYSHLTH'].replace({88:0})
df_selected.PHYSHLTH.unique()

array([15.,  0., 30.,  2., 14., 28., 20.,  7.,  3., 10.,  1.,  5., 17.,
        4., 19.,  6., 12., 25., 27., 21., 22.,  8., 29., 24.,  9., 16.,
       18., 23., 13., 26., 11.])

In [48]:
# 32. var (DIFFWALK)
# Do you have serious difficulty walking or climbing stairs?
# Changing the scale of DIFFWALK to 0,1; No as 0, Yes is 1 
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.DIFFWALK != 7]
df_selected = df_selected[df_selected.DIFFWALK != 9]
df_selected['DIFFWALK'] = df_selected['DIFFWALK'].replace({2:0})
df_selected.DIFFWALK.unique()

array([1., 0.])

In [49]:
# 33. var (CHCCOPD1)
# (Ever told) you have Chronic Obstructive Pulmonary Disease or COPD, emphysema or chronic bronchitis?
# Changing the scale of CHCCOPD1 to 0,1; No as 0, Yes is 1 
# Remove 7, 9 for don't know or refused values

df_selected = df_selected[df_selected.CHCCOPD1 != 7]
df_selected = df_selected[df_selected.CHCCOPD1 != 9]
df_selected['CHCCOPD1'] = df_selected['CHCCOPD1'].replace({2:0})
df_selected.CHCCOPD1.unique()

array([1., 0.])

In [50]:
# 34. var (MAXVO2_)
# Estimated Age-Gender Specific Maximum Oxygen Consumption
# These MAXVO2 are multiplied by 100. So e.g. a MAXVO2 of 2469 is really 24.69
# Remove 99900 for don't know or refused values

df_selected['MAXVO2_'] = df_selected['MAXVO2_'].div(100).round(0)
df_selected = df_selected[df_selected.MAXVO2_ != 99900]
df_selected.MAXVO2_.unique()

array([25., 29., 21., 22., 23., 28., 15., 34., 31., 30., 39., 20., 18.,
       33., 17., 12., 19., 26., 41., 35., 24., 16., 32., 38., 36., 27.,
       37., 40., 13., 48., 14., 44., 45., 10., 42.,  9., 43., 46.,  7.,
       11., 47., 50.,  8., 49.,  6.])

In [51]:
# 35. var (_PACAT1)
# Physical Activity Categories
# _PACAT1 is already ordinal variable on a scale from 1-4 (1 being highly active; 8 being inactive)
# Remove 9 for refused values 

df_selected = df_selected[df_selected._PACAT1 != 9]
df_selected._PACAT1.unique()

array([4., 2., 1., 3.])

In [52]:
# 36. var (_PAINDX1)
# Physical Activity Index (Meet Aerobic Recommendations ?)
# Changing the scale of _PAINDX1 to 0,1; No as 0, Yes is 1
# Remove 9 for refused values 

df_selected = df_selected[df_selected._PAINDX1 != 9]
df_selected['_PAINDX1'] = df_selected['_PAINDX1'].replace({2:0})
df_selected._PAINDX1.unique()

array([0., 1.])

In [53]:
# double chack for NaN
df_selected.isnull().sum(axis = 0)

_MICHD      0
_AGEG5YR    0
SEX         0
EDUCA       0
CHILDREN    0
_TOTINDA    0
VETERAN3    0
EMPLOY1     0
_STATE      0
INCOME2     0
CHECKUP1    0
RENTHOM1    0
MSCODE      0
_PRACE1     0
MARITAL     0
SMOKE100    0
_BMI5       0
_RFBMI5     0
_RFHYPE5    0
TOLDHI2     0
_CHOLCHK    0
CVDSTRK3    0
DIABETE3    0
_FRTLT1     0
_VEGLT1     0
_RFDRHV5    0
HLTHPLN1    0
MEDCOST     0
GENHLTH     0
MENTHLTH    0
PHYSHLTH    0
DIFFWALK    0
CHCCOPD1    0
MAXVO2_     0
_PACAT1     0
_PAINDX1    0
dtype: int64

In [54]:
df_final = df_selected.rename(columns = {'_MICHD':'HeartDiseaseOrAttack', '_AGEG5YR':'Age', 'SEX':'Sex', 
                                         'EDUCA':'Education', 'CHILDREN':'Children','_TOTINDA':'PhysActivity',
                                         'VETERAN3':'Veteran','EMPLOY1':'EmploymentStatus', '_STATE':'State', 
                                         'INCOME2':'Income', 'CHECKUP1':'RoutineCeckup', 'RENTHOM1':'OwnHome', 
                                         'MSCODE':'MSCode','_PRACE1':'Race', 'MARITAL':'MaritalStatus', 
                                         'SMOKE100':'Smoker','_BMI5':'BMI', '_RFBMI5':'OverweightOrObese', 
                                         '_RFHYPE5':'HighBP','TOLDHI2':'HighChol', '_CHOLCHK':'CholCheck',
                                         'CVDSTRK3':'Stroke', 'DIABETE3':'Diabetes', '_FRTLT1':'Fruits', 
                                         '_VEGLT1':'Veggies', '_RFDRHV5':'HvyAlcoholConsump', 
                                         'HLTHPLN1':'AnyHealthcare', 'MEDCOST':'NoDocBcCost', 'GENHLTH':'GenHlth', 
                                         'MENTHLTH':'MentHlth', 'PHYSHLTH':'PhysHlth', 'DIFFWALK':'DiffWalk', 
                                         'CHCCOPD1':'COPD', 'MAXVO2_':'MaxVO2', '_PACAT1':'PhysActive', 
                                         '_PAINDX1':'AerobicRec'})
                                         

In [55]:
# final shape of cleaned dataset
df_final.shape

(138776, 36)

In [56]:
# distribution of heart attacks/diseases vs healty hearts 
df_final.groupby(['HeartDiseaseOrAttack']).size()

HeartDiseaseOrAttack
0.0    123580
1.0     15196
dtype: int64

In [57]:
df_final.head(15)

HeartDiseaseOrAttack   Age  Sex  Education  Children  PhysActivity  \
0                    0.0   9.0  0.0        4.0       0.0           0.0   
1                    0.0   7.0  0.0        6.0       0.0           1.0   
3                    0.0   9.0  0.0        4.0       1.0           0.0   
5                    0.0  11.0  0.0        3.0       0.0           1.0   
6                    0.0  11.0  0.0        5.0       0.0           1.0   
9                    0.0  10.0  1.0        6.0       0.0           1.0   
10                   0.0   9.0  0.0        6.0       0.0           0.0   
15                   0.0  11.0  0.0        4.0       0.0           1.0   
16                   1.0   9.0  0.0        5.0       0.0           0.0   
19                   0.0   8.0  1.0        4.0       0.0           0.0   
21                   0.0  13.0  1.0        6.0       0.0           1.0   
22                   0.0  10.0  0.0        5.0       0.0           0.0   
31                   0.0   7.0  0.0        5.0       2.0           0.0   
33                   0.0  11.0  0.0        4.0       0.0           0.0   
35                   0.0   6.0  0.0        6.0       0.0           1.0   

    Veteran  EmploymentStatus  State  Income  RoutineCeckup  OwnHome  MSCode  \
0       0.0               8.0    1.0     3.0            1.0      1.0     3.0   
1       0.0               3.0    1.0     1.0            4.0      1.0     5.0   
3       0.0               8.0    1.0     8.0            1.0      1.0     3.0   
5       0.0               2.0    1.0     6.0            1.0      1.0     5.0   
6       0.0               7.0    1.0     4.0            1.0      1.0     1.0   
9       0.0               7.0    1.0     8.0            1.0      1.0     1.0   
10      0.0               7.0    1.0     7.0            1.0      1.0     1.0   
15      0.0               5.0    1.0     4.0            1.0      1.0     3.0   
16      0.0               8.0    1.0     1.0            1.0      1.0     3.0   
19      0.0               2.0    1.0     3.0            1.0      1.0     3.0   
21      1.0               7.0    1.0     8.0            1.0      1.0     1.0   
22      0.0               8.0    1.0     1.0            1.0      1.0     3.0   
31      0.0               3.0    1.0     7.0            1.0      1.0     5.0   
33      0.0               7.0    1.0     6.0            1.0      1.0     3.0   
35      0.0               1.0    1.0     8.0            1.0      1.0     5.0   

    Race  MaritalStatus  Smoker   BMI  OverweightOrObese  HighBP  HighChol  \
0    1.0            1.0     1.0  40.0                1.0     1.0       1.0   
1    1.0            2.0     1.0  25.0                1.0     0.0       0.0   
3    1.0            1.0     0.0  28.0                1.0     1.0       1.0   
5    1.0            3.0     0.0  27.0                1.0     1.0       0.0   
6    1.0            3.0     0.0  24.0                0.0     1.0       1.0   
9    1.0            1.0     1.0  25.0                1.0     1.0       1.0   
10   1.0            1.0     1.0  30.0                1.0     1.0       0.0   
15   1.0            3.0     1.0  25.0                1.0     1.0       1.0   
16   1.0            6.0     1.0  30.0                1.0     1.0       1.0   
19   1.0            1.0     0.0  24.0                0.0     0.0       0.0   
21   1.0            1.0     1.0  25.0                0.0     0.0       0.0   
22   1.0            2.0     1.0  34.0                1.0     1.0       1.0   
31   1.0            1.0     1.0  26.0                1.0     0.0       0.0   
33   1.0            1.0     0.0  28.0                1.0     1.0       1.0   
35   1.0            1.0     0.0  33.0                1.0     1.0       0.0   

    CholCheck  Stroke  Diabetes  Fruits  Veggies  HvyAlcoholConsump  \
0         1.0     0.0       0.0     0.0      1.0                0.0   
1         0.0     0.0       0.0     0.0      0.0                0.0   
3         1.0     0.0       0.0     1.0      0.0                0.0   
5    

In [58]:
#************************************************************************************************
df_final.to_csv('heart_disease_health_indicators.csv', sep=",", index=False)
#************************************************************************************************